src/cascadia/TerminalSettingsEditor/Appearances.idl

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import "EnumEntry.idl";
+import "MainPage.idl";
+
+#include "ViewModelHelpers.idl.h"
+
+#define OBSERVABLE_PROJECTED_APPEARANCE_SETTING(Type, Name) \
+    OBSERVABLE_PROJECTED_SETTING(Type, Name);               \
+    Object Name##OverrideSource { get; }
+
+namespace Microsoft.Terminal.Settings.Editor
+{
+    runtimeclass Font : Windows.Foundation.IStringable
+    {
+        String Name { get; };
+        String LocalizedName { get; };
+    }
+
+    runtimeclass AppearanceViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
+    {
+        Windows.Foundation.Collections.IObservableVector<Font> CompleteFontList { get; };
+        Windows.Foundation.Collections.IObservableVector<Font> MonospaceFontList { get; };
```

Comment:
Hm, we would need to give `AppearanceViewModel` a reference to the `ProfileViewModel` then...

Reply:
IIRC `MainPage` passes in some sort of "state" to every new page it navigates to right? Perhaps the font lists (or maybe even some sort of font list manager obj) could be passed into any new profile that's navigated to? LMK if I'm missing a part of the puzzle tho 😄

Reply:
Ah the `ProfileViewModel` already had the font lists, the issue here is how the `Appearances` or `AppearanceViewModel` can get access to the font lists in `ProfileViewModel`. 

Just made a change to give the `Appearances` object a ref to the `ProfileViewModel`, so it grabs the font lists from there now!

Reply:
Yup! `MainPage` can totally pass in the font lists via the `NavigationState`. And that's probably the architecture we'll want. That way, we'll only ever generate the font lists once, and we can reuse them across each `Profiles` page.

The only reason I think we should hold off on that change rn is I think the right approach would be to lazy load them. Like, I'm running into a similar problem with the Actions page right now. If we had `MainPage` load up _all_ of the actions and _all_ of the fonts when you open the SUI, it'll take forever to open it.

Mind filing a task so that we do this later (probably in the v2.0 timeframe)? I think it's worth doing, just not in this PR.

Reply:
I do not think we should continue using `NavigationState` as a bad viewmodel. We need a viewmodel. Navigation state objects are supposed to get temporary information into a page that is necessary _TO NAVIGATE_. If we have a real viewmodel, we don't even need a navigation state class.